### PR TITLE
util: Add EP and domain auth key checking

### PR DESCRIPTION
**Pull request by iziemba**
Verify that user provided auth key hints are sane.

### CODE REVIEW

**Comment by shefty** [comment]
CI failure:
```
18:36:43  - name:   fi_multi_ep -e rdm -v -p "psm3"
18:36:43    timestamp: 20231011-013047+0000
18:36:43    result: Fail
...
18:36:43      fi_getinfo(): functional/multi_ep.c:278, ret=-61 (No data available)
```
I don't know why, might be a problem in fabtests multi_ep

**Comment by shefty** [comment]
fabtests looks okay -- problem looks like it's in the psm3 provider or the definition of auth_key_size needs clarity.

I see psm3 sets the auth_key_size to a non-zero value, but auth_key = null as a default attribute. The man pages suggest that auth_key_size should be set to 0 on return from fi_getinfo() if auth_keys are not used.

**Comment by shefty** [comment]
We may be able to fix this by modifying fi_alter_domain_attr / fi_alter_ep_attr to set the auth_key_size = 0 if auth_key=null.  The provider is setting the auth_key_size to its max value, which allows the checks against hints to work.  But since auth keys aren't being used, the returned fi_info structures should have those cleared.  I'm not sure if that's sufficient, but it may be.  psm3 keys off the auth_key being null when creating resources.

**Comment by j-xiong** [comment]
The psm2 provider is the same.

The original understanding was that the auth_key_size returned from fi_getinfo() indicates the size of the auth_key expected by the provider. The actual auth_key is later set by the application when creating domain or endpoint.

File: prov/util/src/util_attr.c [hunk 1]
```diff
@@ -654,6 +654,23 @@ int ofi_check_domain_attr(const struct fi_provider *prov, uint32_t api_version,
 		return -FI_ENODATA;
 	}
 
+	if (user_attr->auth_key_size &&
+	    (user_attr->auth_key_size != prov_attr->auth_key_size)) {
```
Comment (shefty) [inline prov/util/src/util_attr.c:658]: Based on @j-xiong's comment, lets change this check to
```
if (user_attr->auth_key && user_attr->auth_key_size &&
   (user_attr->auth_key_size != prov_attr->auth_key_size))
```
The below checks may not be needed then

**Pull request merged**